### ISSUE TITLE
False positive on "recur must be in tail position" with `if-some`

### ISSUE THREAD

**Issue by n-hansen** [clj-kondo/clj-kondo#2101]
## version

`clj-kondo v2023.05.26`

## platform

OS X native arm64 executable

## editor

emacs with flycheck-clj-kondo

## problem

When writing `loop`/`recur` with a `if-some` nested within a `if`, and a `recur` in the `some?` leg of the `if-some`, clj-kondo incorrectly flags the `recur` as not being in the tail position.

Note that the outer `if` appears to be necessary to trigger the bug, and the linter correctly handles `if-let` instead of `if-some`.

## repro

CLI repro:
```
$ echo '(loop [] (if true (if-some [_ nil] (recur) 1) 1))' | clj-kondo --lint -
<stdin>:1:36: error: Recur can only be used in tail position.
linting took 6ms, errors: 1, warnings: 0
```

Formatted clj for easier reading:
```clj
(loop []
  (if true
    (if-some [_ nil]
      (recur)
      1)
    1))
```

## expected behavior

There should be no errors reported.

**Comment by borkdude** [comment]
Thanks for the report!
